Controls/laggner

alexlaggner opened the pull request.

alexlaggner committed:
hotfix for remoteAdres bein NaN

alexlaggner committed:
Few Fixes

InvalidArgumentException hinzugefügt und in index statt eines Strings "geworfen"
Try/Catch- Block, damit Server nicht nach Falscheingabe abstürzt
userid muss aus 'data' kommen, nicht props

alexlaggner committed:
validateCommandStructure fix

Nur 'data' zu übergeben würde den Buffer übergeben. Mit dem Buffer werden aufgrund dessen Struktur die 'hasOwnProperty()"-Aufrufe stets false sein

alexlaggner committed:
InCommandCodes fix, Exception add

Fix für Commandunterscheidung: InCommandCodes in GlobalVars weisen nun korrekte JSON-Struktur auf
ValidateCommand wird nun mit richtigem Parameter (konvertiert) aufgerufen und funktioniert
InvalidActionException erstellt und eingebunden

alexlaggner committed:
Angefangen RoomControls zu fixen

static als keyword, da die methode sonst nicht als solche erkennt wird (ich werde nach anderen lösungen suchen)
ES6 Klassen müssen Klassenvariablen im Konstruktor definieren
Weitere Fehlerbehebung folgt.

alexlaggner committed:
createRoom funktioniert nun

als nächstes joinroom: roomkey wird in 'props' übergeben -> input soll immer gleiche struktur aufweisen

alexlaggner committed:
Exception handling erweitert

Exceptions in eigenes File
Neue exceptions für roomcontrols

alexlaggner committed:
bugfixes in roomcontrols

fixed typos which led to errors (JoinRoom -> joinRoom ; roomKey -> roomkey,..)

alexlaggner committed:
added Exceptions to roomcontrols- closeRoom()

alexlaggner committed:
try-catch richtig gesetzt

alexlaggner committed:
export-statement aktualisiert

alexlaggner committed:
Improvements in exceptionhandling, Fixes

Fehlerbehebung in RoomControls
Exceptionhandling in RoomControls
Typo fixes

alexlaggner committed:
closeroom and leaveroom work properly now

Vermutlich gab es ein problem mit dem scope der properties von roomToLeave.
Durch variablendeklarierung wurde dies allerdings behoben. Hier wird noch refactoring nötig sein

alexlaggner committed:
Merge branch 'develop' into controls/laggner

alexlaggner committed:
Further exception handling

extended exception handling
fixed JSON-structures, typos and things that went wrong while rebasing

alexlaggner committed:
createRoom() :mapped socket and userid in room-object

also adapted other functions. ready to implement broadcast method

alexlaggner committed:
broadcastToRoom() implemented and tested

alexlaggner committed:
broadcastToRoom() embedded in joinRoom() and leaveRoom()

alexlaggner committed:
leaveRoomWithSocket added

automatically leaves the room when socket gets closed

alexlaggner committed:
exception handling in startGame()

luko0610 committed:
FIX - Splitte logic and sockets (again). Fixed some other stuff. Functionality, including broadcasts now works.

luko0610 commented:
Exceptions können noch als class definiert werden: https://dev.to/lvidakovic/custom-error-types-in-node-js-491a

alexlaggner reviewed (commented):
wärs sinnvoll im index.js 64-74 im server.on("connect").. zu implementieren? so würde dies schon beim aufbau des sockets passieren, dann ist es garantiert das erste, das passiert.

alexlaggner committed:
Exceptions now extend Error and have adequate subtypes

Exceptions extend Error now
Subtypes for Exceptions created - possible Separation with 'instanceof' possible if needed.
